src/Credentials/CredentialProvider.php

```diff
@@ -224,10 +224,14 @@ public static function memoize(callable $provider)
                         return $creds;
                     }
 
-                    // Refresh expired credentials.
-                    if (!$creds->isExpired()) {
+                    // Check if credentials are expired or will expire within 5 minutes
+                    $expiresWithinThreshold = $creds->getExpiration() - time() < 300;
```

Comment:
nit: I would prefer to have the time window `300` declared as a `const`. 
Also, I am curious to know if this shouldn't be optional?, or somehow configurable?